sprockets/mixins/metrics/statsd.py

```diff
@@ -115,14 +92,124 @@ def on_finish(self):
-                                  (settings['host'], int(settings['port'])))
+    def set_status(self, status_code, reason=None):
+        """Extend tornado `set_status` method to track status code
+        to avoid referencing the _status internal variable
+
+        :param int status_code: Response status code. If ``reason``
+            is ``None``, it must be present in `httplib.responses
+            <http.client.responses>`.
+        :param string reason: Human-readable reason phrase describing
+            the status code. If ``None``, it will be filled in from
+            `httplib.responses <http.client.responses>`.
+        """
+        self.__status_code = status_code
+        super(StatsdMixin, self).set_status(status_code, reason=reason)
+
+
+class StatsDCollector(object):
+    """Collects and submits stats to StatsD via UDP socket.
+
+     This class should be constructed using the
+    :meth:`~sprockets.mixins.statsd.install` method. When installed,
+    it is attached to the :class:`~tornado.web.Application` instance
+    for your web application.
+
+    :param str host: The StatsD host
+    :param str port: The StatsD port
+    :param str namespace: The StatsD bucket to write metrics into.
+    :param bool prepend_metric_type: Optional flag to prepend bucket path
+        with the StatsD metric type
+    :param bool prepend_hostname: Optional flag to prepend bucket path with
+        the hostname
+
+    """
+    METRIC_TYPES = {'c': 'counters',
+                    'ms': 'timers'}
+
+    def __init__(self, host, port, namespace='sprockets',
+                 prepend_metric_type=True, prepend_hostname=True):
+        self._host = host
+        self._port = int(port)
+        self._namespace = namespace
+        self._prepend_metric_type = prepend_metric_type
+        self._prepend_hostname = prepend_hostname
+        self._sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, 0)
+
+    def send(self, path, value, metric_type):
+        """Send a metric to Statsd.
+
+        :param list path: The metric path to record
+        :param mixed value: The value to record
+        :param str metric_type: The metric type
+
+        """
+        msg = '{0}:{1}|{2}'.format(
+            self._build_path(path, metric_type), value, metric_type)
+        try:
+            LOGGER.debug('Sending %s to %s:%s', msg.encode('ascii'),
+                         self._host, self._port)
+            self._sock.sendto(msg.encode('ascii'), (self._host, self._port))
+        except socket.error:
+            LOGGER.exception('Error sending StatsD metrics')
+
+    def _build_path(self, path, metric_type):
+        """Return a normalized path.
+
+        :param list path: elements of the metric path to record
+        :param str metric_type: The metric type
+        :rtype: str
+
+        """
+        path = self._get_prefixes(metric_type) + list(path)
+        return '{}.{}'.format(self._namespace, '.'.join(str(p) for p in path))
```

Comment:
I question whether it's wise to remove the {.} -> {_} logic. Probably not a big deal.